build/EnsureOutputLayout.props

```diff
@@ -19,6 +19,10 @@
     <!-- Use PlatformTarget instead of Platform to keep things more consistent (avoid 'Win32' as a build output) -->
     <BaseOutputPath>$(RepoRoot)\BuildOutput\$(Configuration)\$(PlatformTarget)\</BaseOutputPath>
 
+    <!-- Intermediate files under obj folder. Some intermediate file paths are really long, thus no unnecessary folder added -->
+    <BaseIntermediateOutputPath>$(RepoRoot)\obj\$(Configuration)\$(PlatformTarget)\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
+    <IntermediateOutputPath>$(BaseIntermediateOutputPath)</IntermediateOutputPath>
```

Comment:
What about C++/WinRT's "Generated Files" path? That's spewed directly under each project's directory (and can get disturbingly long). What controls its root, and how do we get it to use a similar path (outside the source trees)?

As a separate PR. But that's the only other artifact I'm aware of generated under source directories (/dev, /test, /installer/dev, /installer/test). It should be possible to put those on read-only media (and build successfully).

Reply:
they are under obj too.